packages/svelte/src/compiler/phases/2-analyze/index.js

```diff
@@ -743,6 +744,13 @@ export function analyze_component(root, source, options) {
 							}
 						])
 					);
+					if (
+						element.type === 'RegularElement' &&
+						element.name.includes('-') &&
```

Comment:
We have a "is_custom_element" (or sth like that) fn somewhere I believe.

Also wondering whether there's a better way, it feels a bit unfortunate to do that, but I also see no other good way if this only happens after traversal... mhm

Reply:
> Also wondering whether there's a better way, it feels a bit unfortunate to do that, but I also see no other good way if this only happens after traversal... mhm

I wondered for a bit but the whole css analysis needs to happen after the normal traversal and we can't know if the class attribute will be added until the css analysis. We might be able to infer something during the transform but it feels way more hacky than this solution.

p.s. i've updated to use the function (but i had to change the return value)